index.bs

```diff
@@ -1812,6 +1821,117 @@ The [=remote end steps=] with |session| and <var ignore>command parameters</var>
 
 </div>
 
+#### The browser.createUserContext Command #### {#command-browser-createUserContext}
+
+The <dfn export for=commands>browser.createUserContext</dfn> command creates a [=user context=] in the user agent.
+
+<dl>
+   <dt>Command Type</dt>
+   <dd>
+      <pre class="cddl remote-cddl">
+      browser.Close = (
+        method: "browser.createUserContext",
+        params: EmptyParams,
```

Comment:
I wonder if we should allow the user to provide an optional name?

Reply:
We currently don't support any names. Does Firefox support names?

Reply:
Firefox containers are named, so I'd expect us to at least return the names for any existing containers. Since (I think) the name has to be unique, we could provide the name as the id. Of course that doesn't work if we enforce UUID (or non-string) ids, or put any other restrictions on the ids.

We don't necessarily have to support setting names, but if we don't (or if we make the names something other than human readable strings), I think gecko would likely end up extending the API to provide some way to communicate how the user contexts map onto containers.

Reply:
> But this could be implemented entirely within the session, right?

Would that mean that other session won't see the names given to contexts by other sessions?

Reply:
Yes, it's true that it depends on what guarantees we want to put in place here. If different sessions should get the same user context id (which would be the default in Firefox, and would make sense if you assume that user contexts can also be created outside of WebDriver) then we need it to be a property of the implementation. If different sessions should get the different user context ids (which might be easier to implement in Chrome) then the ids can be properties of the session alone.

In practice I think it would be trivial for different sessions to map user context ids, especially if browsing context ids are shared (which I think is an open discussion, and which worries me a bit, but which the spec currently requires).

In Firefox, given that the underlying container name is exposed in the UI, I'd be very surprised if we don't provide _some_ way to get at that name, even if we also layer on a per-session id on top of that.

Reply:
I think we are aiming at "different sessions should get the same user context id" as it's also how it's implemented in Chrome. So I think it's the same in both browsers. With the names it'd different because the WebDriver BiDi implementation in Chrome is not part of the browser. And theoretically multiple BiDi implementation instances could talk to the same Chrome instance and they will only have access to the data that lives in the Chrome instance and will have different session data. 

Do user context names in Firefox have to be unique? Do you also have a unique ID or do you plan to use the name as the ID?

Reply:
Based on experiments, the container name in Firefox doesn't have to be unique, so I guess there's also an underlying id that's different.

Reply:
Yes, [each context has its own context id](https://searchfox.org/mozilla-central/rev/77aeaa184d59944ecb117e452ea43f5bd1556e4c/toolkit/components/contextualidentity/ContextualIdentityService.sys.mjs#264).

Reply:
As I understand we leave out the names for now and might add it eventually?